### PR TITLE
SAGA: Support Chinese IHNM version

### PR DESCRIPTION
**Pull request by phcoder**

### CODE REVIEW

File: engines/saga/font.cpp [hunk 1]
```diff
@@ -196,6 +203,120 @@ void DefaultFont::loadChineseFontITE(const Common::String& fileName) {
...
+	// No idea what is the beginning, some 3 values and then some bitmask,
+	// anyway file is constant and as long as we know how to interpret
+	// and match glyphs we're good
+	int curIdx = 1286;
+
+	// It's just sequential big5 codepoints by compartments specified in
+	// Big5 specification. Compartments are out of order
+	// 0x8140 to 0xA0FE	Reserved for user-defined characters 造字
+	// Not present
+	// 0xA440 to 0xC67E	Frequently used characters 常用字
```
Comment (bluegr) [inline engines/saga/font.cpp:254]: Please, rephrase this, it's a bit confusing the way it's written

Comment (sev-) [reply engines/saga/font.cpp:254]: Interestingly, it looks quite clear to me. 0x8140 to 0xA0FE	range used by that compartment of Big5 is not present in the font, while the "Frequently used characters" compartment 0xA440 to 0xC67E  is present.

How would you suggest to rephrase it, @bluegr ?

Comment (bluegr) [reply engines/saga/font.cpp:254]: For example:

         // Big5 specification. Compartments are out of order
	// - 0x8140 to 0xA0FE	Reserved for user-defined characters 造字 - Not present
	// - 0xA440 to 0xC67E	Frequently used characters 常用字

File: engines/saga/font.cpp [hunk 2]
```diff
@@ -150,7 +154,10 @@ DefaultFont::DefaultFont(SagaEngine *vm) : Font(vm), _fontMapping(0), _chineseFo
 		_fonts[i].outline.font = NULL;
 		_fonts[i].normal.font = NULL;
 #endif
-		loadFont(&_fonts[i],	_vm->getFontDescription(i)->fontResourceId);
+		if (i == 0 && _vm->getGameId() == GID_IHNM && _vm->getLanguage() == Common::ZH_TWN)
+			loadChineseFontIHNM(&_fonts[i],	_vm->getFontDescription(i)->fontResourceId);
+		else
+			loadFont(&_fonts[i],	_vm->getFontDescription(i)->fontResourceId);
```
Comment (mgerhardy) [inline engines/saga/font.cpp:160]: a tabulator sneaked in here

**Comment by bluegr** [comment]
@phcoder Can you please rebase?

**Comment by phcoder** [comment]
> @phcoder Can you please rebase?

Done

File: engines/saga/font.cpp (hunk 1 above)
Comment (bluegr) [reply engines/saga/font.cpp:281]: Looking at this again, this looks good

**Comment by bluegr** [comment]
Well done, thanks!

**Pull request merged**